docs/hugo/content/docs/authentication.md

```diff
@@ -0,0 +1,95 @@
+# Authentication in Azure Service Operator v2
+
+Azure Service Operator supports two different styles of authentication today.
+
+1. Service Principal
+2. aad-pod-identity authentication (managed identity)
+
+## Service Principal
+
+### Prerequisites
+1. An existing Azure Service Principal.
+
+To use Service Principal authentication, specify an `aso-controller-settings` secret with `AZURE_CLIENT_SECRET` set.
+`AZURE_CLIENT_ID` must be set to the Service Principal client ID.
+`AZURE_CLIENT_SECRET` must be set to the Service Principal client secret.
+
+```bash
+cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: Secret
+metadata:
+ name: aso-controller-settings
+ namespace: azureserviceoperator-system
+stringData:
+ AZURE_SUBSCRIPTION_ID: "$AZURE_SUBSCRIPTION_ID"
+ AZURE_TENANT_ID: "$AZURE_TENANT_ID"
+ AZURE_CLIENT_ID: "$AZURE_CLIENT_ID"
+ AZURE_CLIENT_SECRET: "$AZURE_CLIENT_SECRET"
+EOF
+```
+
+## Managed Identity (aad-pod-identity)
+
+### Prerequisites
+1. An existing Azure Managed Identity.
+2. [aad-pod-identity](https://github.com/Azure/aad-pod-identity) installed into your cluster. If you are running ASO on an Azure Kubernetes Service (AKS) cluster, you can instead use the
+   [integrated aad-pod-identity](https://docs.microsoft.com/azure/aks/use-azure-ad-pod-identity).
+
+First, set the following environment variables:
+```bash
+export IDENTITY_RESOURCE_GROUP="..."
+export IDENTITY_NAME="..."
+export AZURE_SUBSCRIPTION_ID="..."
+export AZURE_TENANT_ID="..."
```

Comment:
Same comment here - one of the regular trip hazards I find is when documentation is unclear about what's expected. 

E.g. is `IDENTIY_RESOURCE_GROUP` just the name of the group (is it case sensitive?), or a fully qualified path including the subscription ID, or maybe an ARM URL?

Reply:
I've provided examples here and also added some comments at the end of these to explain what they are.